packages-content-model/roosterjs-content-model-dom/lib/modelApi/creators/createTableCell.ts

```diff
@@ -27,6 +29,6 @@ export function createTableCell(
         spanLeft,
         spanAbove,
         isHeader: !!isHeader,
-        dataset: {},
+        dataset: dataset ?? {},
```

Comment:
clone this object rather than just directly assign it:

dataset: {...dataset}

Otherwise two cells will share the same dataset object, modifying one of them will cause the other one also got changed.